roboteam_ai/src/stp/plays/defensive/DefendPass.cpp

```diff
@@ -148,6 +151,32 @@ void DefendPass::calculateInfoForOffenders() noexcept {
     }
 }
 
+void DefendPass::calculateInfoForHarasser() noexcept {
+    auto enemyClosestToBall = world->getWorld()->getRobotClosestToBall(world::them);
+
+    if (!stpInfos["harasser"].getRobot() || !enemyClosestToBall) {
+        stpInfos["harasser"].setPositionToMoveTo(world->getWorld()->getBall()->get()->position);
+        return;
+    }
+
+    auto ballPos = world->getWorld()->getBall()->get()->position;
+    auto robotToBallAngle = (ballPos - stpInfos["harasser"].getRobot()->get()->getPos()).toAngle();
+    auto ballToEnemyAngle = (enemyClosestToBall->get()->getPos() - ballPos).toAngle();
+    auto angleDiff = robotToBallAngle.shortestAngleDiff(ballToEnemyAngle);
+    if (angleDiff > M_PI / 3.0) {  // If the enemy is between us and the ball, dont go to the ball directly but further away, to avoid crashing
+        auto enemyPos = enemyClosestToBall->get()->getPos();
+        auto targetPos = FieldComputations::projectPointToValidPositionOnLine(field, enemyPos + (ballPos - enemyPos).stretchToLength(0.50), enemyPos,
+                                                                              enemyPos + (ballPos - enemyPos).stretchToLength(10), AvoidObjects{}, 0.0,
+                                                                              control_constants::ROBOT_RADIUS * 2, 0.0);
+        stpInfos["harasser"].setPositionToMoveTo(targetPos);
+        stpInfos["harasser"].setAngle((enemyPos - ballPos).angle());
+    } else {
+        stpInfos["harasser"].setShouldAvoidTheirRobots(false);  // Allow the harasser to get close to the enemy robot by not caring about collisions with enemy robots
```

Comment:
this will only trigger once there is no enemy robots directly between the harasser and the ball. Normally our robots keep a certain distance from the enemy robots to avoid crashing but this means that we won't be able to take the ball. If there are no enemy robots between us and the ball we don't need to worry about collisions so we can just move straight toward the ball